src/sage/plot/graphics.py

```diff
@@ -36,6 +36,8 @@
 # ****************************************************************************
 
 import os
+import contextlib
+import numbers
```

Comment:
This could probably be moved local to the function to avoid the import until someone needs it. But only if `matplotlib()` is not called in a tight loop.

Reply:
`numbers` is most probably already loaded, we just want to bring `numbers.Integral` in scope so I replaced it with `from numbers import Integral`. I don't see any difference between that and the next line `from collections.abc import Iterable`, etc.

As your other suggestion, I'm not importing contextlib anymore so it's only that one.

Reply:
Ok, thanks, it's not too important. I know that new top-level imports can sometimes slow down sage at startup but it's something that can easily be profiled and refactored later if it is really an issue.